File: algorithm/src/main/java/baekjoon/p02484/Main.java
package baekjoon.p02484;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[][] inputs = new int[n][4];

        for (int i = 0; i < inputs.length; i++) {
            for (int j = 0; j < inputs[i].length; j++) {
                inputs[i][j] = sc.nextInt();
            }
        }

        System.out.println(solve(inputs));
    }

    private static int solve(int[][] inputs) {
        int ret = -1;

        for (int[] input : inputs) {
            ret = Math.max(ret, score(input));
        }

        return ret;
    }

    private static int score(int[] input) {
        int[] cnt = new int[7];
        for (int n : input) {
            cnt[n]++;
        }

        for (int i = 1; i <= 6; i++) {
            if (cnt[i] == 4) {
                return (50000 + i * 5000);
            } else if (cnt[i] == 3) {
                return (10000 + i * 1000);
            }
        }

        int[] cnt2 = find(cnt, 2);
        if (cnt2.length == 2) {
            return (2000 + cnt2[0] * 500 + cnt2[1] * 500);
        } else if (cnt2.length == 1) {
            return (1000 + cnt2[0] * 100);
        } else {
            return max(input) * 100;
        }
    }

    private static int[] find(int[] cnt, int target) {
        List<Integer> list = new ArrayList<>();

        for (int i = 1; i <= 6; i++) {
            if (cnt[i] == target) {
                list.add(i);
            }
        }

        int[] ret = new int[list.size()];
        for (int i = 0; i < ret.length; i++) {
            ret[i] = list.get(i);
        }

        return ret;
    }

    private static int max(int[] input) {
        int ret = 0;

        for (int n : input) {
            ret = Math.max(ret, n);
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p02484/Main.kt
package baekjoon.p02484

fun main(args: Array<String>) {
    val n = readLine()!!.toInt()
    val inputs = Array(n) { readLine()!!.split(" ").map { it.toInt() } }
    println(solve(inputs))
}

private fun solve(inputs: Array<List<Int>>): Int {
    return (0..inputs.lastIndex).map { score(inputs[it]) }.max()!!
}

private fun score(input: List<Int>): Int {
    val sorted = input.sorted()
    return when (sorted.distinct().size) {
        1 -> (50000 + input[0] * 5000)
        2 -> {
            when (sorted[0] == sorted[1] && sorted[2] == sorted[3]) {
                true -> (2000 + sorted[0] * 500 + sorted[2] * 500)
                false -> (10000 + sorted[1] * 1000)
            }
        }
        3 -> {
            (1000 + when (sorted[0] == sorted[1]) {
                true -> sorted[0]
                false -> sorted[2]
            } * 100)
        }
        else -> (input.max()!! * 100)
    }
}